customer-commons/src/main/java/no/unit/nva/customer/model/ApplicationDomain.java

```diff
@@ -0,0 +1,40 @@
+package no.unit.nva.customer.model;
+
+import static java.util.Objects.isNull;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.net.URI;
+import java.util.Arrays;
+import nva.commons.core.SingletonCollector;
+
+public enum ApplicationDomain {
+
+    NVA(URI.create("nva.unit.no"));
+    private final URI uri;
+
+    @JsonCreator
+    ApplicationDomain(URI uri) {
+        this.uri = uri;
+    }
+
+    public static ApplicationDomain fromUri(URI candidate) {
+        var uri = mapValuesFromPreviousDatamodel(candidate);
+        return Arrays.stream(ApplicationDomain.values())
+                   .filter(applicationDomain -> applicationDomain.getUri().equals(uri))
+                   .collect(SingletonCollector.collect());
+    }
+
+    @JsonValue
+    @Override
+    public String toString() {
+        return this.uri.toString();
```

Comment:
Either uri.toString(), or getUri().toString(). On the whole, I'd prefer the latter — I'm not sure this is entirely necessary, but if we imagine that we might decide to do something in the accessor — for example adding the HTTPS scheme, we avoid having to duplicate the change in the values.

Reply:
fixed